py/grammar.h

```diff
@@ -239,15 +253,27 @@ DEF_RULE(term_op, nc, or(4), tok(OP_STAR), tok(OP_SLASH), tok(OP_PERCENT), tok(O
 DEF_RULE(factor, nc, or(2), rule(factor_2), rule(power))
 DEF_RULE(factor_2, c(factor_2), and(2), rule(factor_op), rule(factor))
 DEF_RULE(factor_op, nc, or(3), tok(OP_PLUS), tok(OP_MINUS), tok(OP_TILDE))
+#if MICROPY_ASYNC_SYNTAX
+DEF_RULE(power, c(power), and(2), rule(await_or_atom_expr), opt_rule(power_dbl_star))
+DEF_RULE(await_or_atom_expr, nc, or(2), rule(await_expr), rule(atom_expr))
+DEF_RULE(await_expr, c(await_expr), and(3), tok(KW_AWAIT), rule(atom), opt_rule(atom_expr_trailers))
+DEF_RULE(atom_expr, c(atom_expr), and(2), rule(atom), opt_rule(atom_expr_trailers))
+DEF_RULE(atom_expr_trailers, c(atom_expr_trailers), one_or_more, rule(trailer))
+#else
 DEF_RULE(power, c(power), and(3), rule(atom), opt_rule(power_trailers), opt_rule(power_dbl_star))
 DEF_RULE(power_trailers, c(power_trailers), one_or_more, rule(trailer))
+#endif
```

Comment:
In order not to have lots of #if's in the compiler, I would make the non-async version of the grammar use the same rule names as the async version, and just skip the rule that has the "await" keyword.